### PR TITLE
Code redundancy fixes pass 2

### PR DESCRIPTION
**Pull request by xtqqczze**
_All changes have been merged._

Follow-up to #12916

* <strike>Autofix [RCS1036](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1036.md): Remove redundant empty line</strike> <ins>changes split to #13404</ins>
* <strike>Autofix [RCS1033](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1033.md): Remove redundant boolean literal</strike> <ins>changes split to #13454, #13994</ins>
* <strike>Autofix [RCS1114](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1114.md): Remove redundant delegate creation</strike> <ins>changes merged in #13441</ins>

### CODE REVIEW

**Comment by xtqqczze** [comment]
rebased to restart CI

File: src/System.Management.Automation/engine/parser/SemanticChecks.cs [hunk 1]
```diff
@@ -348,7 +348,7 @@ internal static void CheckArrayTypeNameDepth(ITypeName typeName, IScriptExtent e
         {
             int count = 0;
             ITypeName type = typeName;
-            while ((type is TypeName) == false)
+            while ((!(type is TypeName)))
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/parser/SemanticChecks.cs:351]:
```suggestion
            while (type is not TypeName)
```

File: src/System.Management.Automation/namespaces/RegistryProvider.cs [hunk 2]
```diff
@@ -3061,7 +3061,7 @@ private bool HasRelativePathTokens(string path)
 
                     if (
                         expandAll ||
-                        ((Context.SuppressWildcardExpansion == false) && (valueNameMatcher.IsMatch(valueNameToMatch))) ||
+                        ((!Context.SuppressWildcardExpansion) && (valueNameMatcher.IsMatch(valueNameToMatch))) ||
                        ((Context.SuppressWildcardExpansion) && (string.Equals(valueNameToMatch, requestedValueName, StringComparison.OrdinalIgnoreCase))))
```
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/RegistryProvider.cs:3065]: Extra parentheses.

File: src/System.Management.Automation/engine/parser/SemanticChecks.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/parser/SemanticChecks.cs:351]: change is in #13338

File: src/System.Management.Automation/namespaces/RegistryProvider.cs (hunk 2 above)
Comment (xtqqczze) [reply src/System.Management.Automation/namespaces/RegistryProvider.cs:3065]: I will fix this in seperate PR using [RCS1032](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1032.md): Remove redundant parentheses

Comment (xtqqczze) [reply src/System.Management.Automation/namespaces/RegistryProvider.cs:3065]: Actually neither RCS1032 nor IDE0047 classify these parentheses as unecessary.

Comment (xtqqczze) [reply src/System.Management.Automation/namespaces/RegistryProvider.cs:3065]: Anyway I would prefer remove parentheses in seperate PR.

**Review by iSazonov — APPROVED** [review]

**Comment by TravisEz13** [comment]
Marked for maintainer review because of large line changes.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs [hunk 3]
```diff
@@ -193,7 +187,6 @@ public override string ReadLine()
         /// <exception cref="PipelineStoppedException">
         /// If Ctrl-C is entered by user
         /// </exception>
-
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs:196]: This type of change is not a code redundancy change.  Please don't combine.  The PR is too large to review already.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs [hunk 4]
```diff
@@ -2831,7 +2831,7 @@ internal static bool QueryServiceConfig(NakedWin32Handle hService, out NativeMet
                 cbBufSize: 0,
                 pcbBytesNeeded: out bufferSizeNeeded);
 
-            if (status != true && Marshal.GetLastWin32Error() != ERROR_INSUFFICIENT_BUFFER)
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs:2834]: This also isn't code redundancy, please try to fix one issue in a PR, and limit the size of a PR.

Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs:2834]: http://dev2ops.org/2012/03/devops-lessons-from-lean-small-batches-improve-flow/
https://cloud.google.com/solutions/devops/devops-process-working-in-small-batches

**Review by TravisEz13 — COMMENTED** [review]
I'm not going to say to change anything, because I think that this PR is not viable.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs (hunk 3 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs:196]: It is a redundant empty line. I can split into seperate PR.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs (hunk 4 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs:2834]: It is a redundant boolean literal. I can split into seperate PR.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs (hunk 4 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs:2834]: @xtqqczze In future it is better to make one fix per a PR but current I reviewed the PR twice :-) so please do not break a commit history.

Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs:2834]: @TravisEz13 @iSazonov I was thinking about seperating 02f4b96 into a new PR, but I would rather not break the commit history again.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowCommand/ShowCommand.cs [hunk 5]
```diff
@@ -278,8 +278,8 @@ private void RunScriptSilentlyAndWithErrorHookup(string script)
             // errors are not created here, because there is a field for it used in the final pop up
             PSDataCollection<object> output = new PSDataCollection<object>();
 
-            output.DataAdded += new EventHandler<DataAddedEventArgs>(this.Output_DataAdded);
-            _errors.DataAdded += new EventHandler<DataAddedEventArgs>(this.Error_DataAdded);
+            output.DataAdded += this.Output_DataAdded;
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowCommand/ShowCommand.cs:281]: This still seems like a different type of fix to me.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

**Comment by TravisEz13** [comment]
I'm leaving the review-maintainer tag, so the maintainers can review what was done here and see if we should update the contribution guide.

**Comment by iSazonov** [comment]
> limit PR size

It seems we already have this recommendation. Also for automatic fixes (by script or analyzer) it is better to fix all code base at a time.

**Comment by TravisEz13** [comment]
> It seems we already have this recommendation. Also, for automatic fixes (by script or analyzer) it is better to fix all code base at a time.

Then, I'm at a deadlock.  Large fixes are not reasonably reviewable.

**Comment by xtqqczze** [comment]
OK, I will seperate RCS1114 changes into new PR, so that they can be reviewed and merged. I will not rebase this PR, leaving @iSazonov approval intact. This means we will have to merge in order.

**Comment by iSazonov** [comment]
> Then, I'm at a deadlock. Large fixes are not reasonably reviewable.

We have already gained some experience in the area:
- do only one kind of a change per PR
- combine  changes in some small commits and review by commit.
- pull PRs per module/project, for SMA - per subsystem.

**Comment by xtqqczze** [comment]
@iSazonov @TravisEz13 I have split some changes from `RCS1033: Remove redundant boolean literal` to #13454 based upon project.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowCommand/ShowCommand.cs (hunk 5 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowCommand/ShowCommand.cs:281]: These changes are now presented in #13441.

**Comment by daxian-dbw** [comment]
@PowerShell/powershell-maintainers reviewed this PR and agreed to accept it for 7.2-preivew. So we should delay merging it until after 7.1-RC1 release branch gets created.
> The changes make sense and the chance of confliction is relatively low, but it requires line-by-line review to make sure there is no potential incorrect change, so it's better to delay merging it till 7.2 preview to avoid possible regression.

**Comment by iSazonov** [comment]
Set Draft till 7.2 Preview1

**Comment by iSazonov** [comment]
@xtqqczze Is it still Draft?

**Comment by xtqqczze** [comment]
rebased

**Comment by xtqqczze** [comment]
54e09b3 removes unnecessary parenthesis

**Comment by xtqqczze** [comment]
remaining CodeFactor issues are pre-existing

**Comment by iSazonov** [comment]
@xtqqczze Please resolve merge conflicts.

**Comment by iSazonov** [comment]
@TravisEz13 Please update your review.

**Comment by xtqqczze** [comment]
@iSazonov I'm minded to close this in favour of #13994, which should include all of these changes. This PR has been open for many months, I think a fresh start would be easier to review.

**Comment by xtqqczze** [comment]
all changes are merged, closed

**Pull request closed without merging**